### PR TITLE
create load scaling

### PR DESCRIPTION
**Pull request by bendaamerahm**
Signed-off-by: bendaamerahm <ahmed.bendaamer@rte-france.com>

### CODE REVIEW

File: src/translations/network-modifications-en.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "network_modifications/LINE_SPLIT_WITH_VOLTAGE_LEVEL": "Splitting a line {computedLabel}",
     "network_modifications/LINE_ATTACH_TO_VOLTAGE_LEVEL": "Attaching line {computedLabel}",
     "network_modifications/LINES_ATTACH_TO_SPLIT_LINES": "Attaching lines to splitting lines {computedLabel}",
+    "network_modifications/LOAD_SCALING": "Load scalable {computedLabel}",
```
Comment (FranckLecuyer) [inline src/translations/network-modifications-en.json:9]: Load scaling

File: src/translations/fr.json [hunk 2]
```diff
@@ -497,6 +497,7 @@
     "ReplacingLines": "Liaisons de remplacement",
     "LineDivisionError": "Erreur lors de la création d'une coupure",
     "LineAttachmentError": "Erreur lors de la création d'un piquage",
+    "LoadScalingError": "Erreur lors de la création de Variation plan de consommation",
```
Comment (FranckLecuyer) [inline src/translations/fr.json:500]: Erreur lors de la création de la variation du plan de consommation

File: src/translations/fr.json [hunk 3]
```diff
@@ -663,6 +665,16 @@
     "ConnectionDirection": "Sens départ",
     "Top": "Haut",
     "Bottom": "Bas",
+    "DeltaP": "Δ P",
+    "TargetPText": "P cible",
+    "CreateVariation": "Ajouter Variation",
+    "ActiveVariationMode": "Mode variation de l'actif",
+    "ReactiveVariationMode": "Mode variation du réactif",
+    "RegularDistribution": "Équipartition",
+    "TanFixed": "tan(φ) fixe",
```
Comment (FranckLecuyer) [inline src/translations/fr.json:674]: TanPhiFixed

File: src/translations/en.json [hunk 4]
```diff
@@ -663,6 +664,16 @@
     "ConnectionDirection": "Connection direction",
     "Top": "Top",
     "Bottom": "Bottom",
+    "DeltaP": "Δ P",
+    "TargetPText": "Target P",
+    "CreateVariation": "Add Variation",
+    "ActiveVariationMode": "Active variation mode",
+    "ReactiveVariationMode": "Reactive variation mode",
+    "RegularDistribution": "Regular distribution",
+    "TanFixed": "tan(φ) fixed",
```
Comment (FranckLecuyer) [inline src/translations/en.json:673]: TanPhiFixed

File: src/translations/en.json [hunk 5]
```diff
@@ -663,6 +664,16 @@
...
+    "ConstantQWithoutUnit": "Constant Q",
+    "Variations": "Variations",
+    "LoadScaling": "Load scalaing",
```
Comment (FranckLecuyer) [inline src/translations/en.json:676]: "Load scaling"

File: src/components/network/constants.js [hunk 6]
```diff
@@ -64,6 +64,22 @@ export const REACTIVE_LIMIT_TYPES = [
...
+
+export const ACTIVE_VARIATION_MODE = [
+    { id: 'PROPORTIONAL', label: 'Proportional' },
+    { id: 'REGULAR_DISTRIBUTION', label: 'RegularDistribution' },
+    { id: 'VENTILATION', label: 'Ventilation' },
+];
+
+export const REACTIVE_VARIATION_MODE = [
+    { id: 'CONSTANT_Q', label: 'ConstantQ' },
+    { id: 'TAN_FIXED', label: 'TanFixed' },
```
Comment (FranckLecuyer) [inline src/components/network/constants.js:80]: TAN_PHI_FIXED

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/components/dialogs/load-scaling-dialog.js [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+        margin: 3,
+        maxWidth: 200,
+    },
+    padding: {
+        padding: '15px',
+    },
+}));
+
+const ACTIVE_VAR_MODE_DEFAULT_VALUE = 'PROPORTIONAL';
+const REACTIVE_VAR_MODE_DEFAULT_VALUE = 'TAN_FIXED';
```
Comment (FranckLecuyer) [inline src/components/dialogs/load-scaling-dialog.js:62]: TAN_PHI_FIXED

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Tests : 
Some active variation mode values and some reactive variation mode values are not completely visible in language EN

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/components/graph/menus/modification-list-item.js [hunk 8]
```diff
@@ -73,6 +73,8 @@ export const ModificationListItem = ({
             return modif.lineToAttachToId;
         } else if (modif.type === 'LINES_ATTACH_TO_SPLIT_LINES') {
             return modif.attachedLineId;
+        } else if (modif.type === 'LOAD_SCALING') {
+            return modif.uuid;
```
[the remark this answers is not in the document]
Comment (bendaamerahm) [reply src/components/graph/menus/modification-list-item.js:77]: to check with our PO

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Tests:
Adding or deleting a filter does not enabling the 'Validate' button

File: src/components/graph/menus/modification-list-item.js [hunk 9]
```diff
@@ -71,6 +71,8 @@ export const ModificationListItem = ({
             return modif.lineToAttachToId;
         } else if (modif.type === 'LINES_ATTACH_TO_SPLIT_LINES') {
             return modif.attachedLineId;
+        } else if (modif.type === 'LOAD_SCALING') {
```
Comment (FranckLecuyer) [inline src/components/graph/menus/modification-list-item.js:74]: remove this case : no uuid will be appended to generic load scaling label

**Review by FranckLecuyer — APPROVED** [review]

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
See remaining remarks

**Review by FranckLecuyer — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**